File: leetcode/medium/119.Pascal's Triangle/python/pascalTriangle.py
"""Pascal's Triangle"""

def pascal_triangle(rowIndex):
    """Given an integer rowIndex, return the rowIndexth (0-indexed) row of the Pascal's triangle."""
    row_pascal = [1]
    for i in range(rowIndex):
        n = len(row_pascal)
        row_pascal.insert(0, 0)
        row_pascal.append(0)
        for j in range(n):
            value = row_pascal[j] + row_pascal[j + 1]
            row_pascal[j] = value
        del row_pascal[-1]
    return row_pascal

# Testing
row_index0 = 3
row_index1 = 0
row_index2 = 1
print(pascal_triangle(row_index2))

'''Complexity Analysis:
Time Complexity: O(rowIndex2)
Space Complexity: O(rowIndex)'''

File: leetcode/easy/119.Pascal's Triangle/js/pascalTriangle.js
"use strict";

// Pascal's Triangle

/**
 * @param {int} rowIndex
 * @return {Array int} 
 * @author {Alex Soto}
 */

var pascalTriangle = function(rowIndex) {
    let rowPascal = [1];
    for (let i = 0; i < rowIndex; i++) {
        let n = rowPascal.length;
        rowPascal.unshift(0);
        rowPascal.push(0);
        for (let j = 0; j < n; j++) {
            rowPascal[j] = rowPascal[j] + rowPascal[j + 1];
        };
        rowPascal.splice(-1, 1);
    };
    return rowPascal;
};

// Testing
let rowIndex0 = 3;
let rowIndex1 = 0;
let rowIndex2 = 1;
console.log(pascalTriangle(rowIndex0));

// Complexity Analysis:
// Time Complexity: O(rowIndex2)
// Space Complexity: O(rowIndex)